File: demos/03-semantic-kernel/07-agents/email-agent/Plugins/EmailPlugin.cs
using System.ComponentModel;
using Microsoft.SemanticKernel;

public class EmailPlugin
{
    [KernelFunction("send_email")]
    [Description("Sends an email to a recipient.")]
    public Task SendEmailAsync(
        Kernel kernel,
        List<string> recipientEmails,
        string subject,
        string body
    )
    {
        Console.WriteLine("Mocking e-mail sent!");
        return Task.CompletedTask;
    }
}

File: demos/03-agent-service/06-semantic-kernel/sk-ai-agent-py/email_plugin.py
from semantic_kernel.functions import kernel_function
from typing import Annotated

class EmailPlugin:
    """A Plugin to simulate email functionality."""
    
    @kernel_function(description="Sends an email.")
    def send_email(self,
                to: Annotated[str, "Who to send the email to"],
                subject: Annotated[str, "The subject of the email."],
                body: Annotated[str, "The text body of the email."]):
        print("\nTo:", to)
        print("Subject:", subject)
        print(body, "\n")
